Snail/Subscriber.swift

```diff
@@ -2,14 +2,22 @@
 
 import Foundation
 
-public class Subscriber<T> {
+public class Subscriber<T>: DisposableType {
     let queue: DispatchQueue?
     let handler: (Event<T>) -> Void
-    weak var observable: Observable<T>?
+    public weak var observable: Observable<T>?
 
     public init(queue: DispatchQueue?, observable: Observable<T>, handler: @escaping (Event<T>) -> Void) {
         self.queue = queue
         self.handler = handler
         self.observable = observable
     }
+
+    public func dispose() {
+        observable?.removeSubscriber(subscriber: self)
+    }
+
+    public func set(on disposer: Disposer) {
```

Comment:
can this be `add(disposer: Disposer)`

Reply:
`add(to:)` and `set(on:)` were my favorites since they represent the _inverted_ addTo.